internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -644,6 +646,47 @@ func (r *AgentServiceConfigReconciler) newImageServiceRoute(ctx context.Context,
 	return route, mutateFn, nil
 }
 
+func (r *AgentServiceConfigReconciler) newInsecureIPXERoute(ctx context.Context, log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) (client.Object, controllerutil.MutateFn, error) {
+	weight := int32(100)
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-ipxe", imageServiceName),
+			Namespace: r.Namespace,
+		},
+	}
+	routeSpec := routev1.RouteSpec{
+		Path: imageservice.BootArtifactsPath,
+		To: routev1.RouteTargetReference{
+			Kind:   "Service",
+			Name:   imageServiceName,
+			Weight: &weight,
+		},
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.FromString(imageServiceName),
+		},
+		WildcardPolicy: routev1.WildcardPolicyNone,
+		TLS:            &routev1.TLSConfig{InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyAllow},
```

Comment:
Does this work?

The pod is serving https, so we've been using reencrypt for the other routes and I couldn't get it to work without changing the service in my tests.

Reply:
Okay, I think this needs to be WIP then, because we'd also need to add the insecure port to the image-service service, right?